pkg/file/seekjoiner/internal/job.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package internal
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"io"
+
+	"github.com/ethersphere/bee/pkg/file"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+type SimpleJoinerJob struct {
+	addr       swarm.Address
+	rootData   []byte
+	spanLength int64
+	off        int64
+	levels     int
+
+	ctx    context.Context
+	getter storage.Getter
+}
+
+// NewSimpleJoinerJob creates a new simpleJoinerJob.
+func NewSimpleJoinerJob(ctx context.Context, getter storage.Getter, rootChunk swarm.Chunk) *SimpleJoinerJob {
+	// spanLength is the overall  size of the entire data layer for this content addressed hash
+	spanLength := binary.LittleEndian.Uint64(rootChunk.Data()[:8])
+	levelCount := file.Levels(int64(spanLength), swarm.SectionSize, swarm.Branches)
+	j := &SimpleJoinerJob{
+		addr:       rootChunk.Address(),
+		ctx:        ctx,
+		getter:     getter,
+		spanLength: int64(spanLength),
+		rootData:   rootChunk.Data()[8:],
+		levels:     levelCount,
+	}
+
+	return j
+}
+
+// Read is called by the consumer to retrieve the joined data.
+// It must be called with a buffer equal to the maximum chunk size.
+func (j *SimpleJoinerJob) Read(b []byte) (n int, err error) {
+	read, err := j.ReadAt(b, j.off)
+	if err != nil && err != io.EOF {
+		return read, err
+	}
+
+	j.off += int64(read)
+	return read, err
+}
+
+func (j *SimpleJoinerJob) ReadAt(b []byte, off int64) (read int, err error) {
+	return j.readAtOffset(b, j.rootData, 0, j.spanLength, off)
+}
+
+func (j *SimpleJoinerJob) readAtOffset(b, data []byte, cur, subTrieSize, off int64) (read int, err error) {
+	if off >= j.spanLength {
+		return 0, io.EOF
+	}
+
+	if subTrieSize <= int64(len(data)) {
+		capacity := int64(cap(b))
+		dataOffsetStart := off - cur
+		dataOffsetEnd := dataOffsetStart + capacity
+
+		if lenDataToCopy := int64(len(data)) - dataOffsetStart; capacity > lenDataToCopy {
+			dataOffsetEnd = dataOffsetStart + lenDataToCopy
+		}
+
+		bs := data[dataOffsetStart:dataOffsetEnd]
+		n := copy(b, bs)
+		return n, nil
+	}
+
+	for cursor := 0; cursor < len(data); cursor += swarm.SectionSize {
```

Comment:
so this wont handle variable references yet.
btw section size is defo the wrong name here

Reply:
this is the same like in the previous implementation. it does not support variable references because it also does not support encryption. once the latter is supported then the former will be too